packages/base/src/stacBrowser/hooks/useStacSearch.ts

```diff
@@ -0,0 +1,292 @@
+import { IJGISLayer, IJupyterGISModel } from '@jupytergis/schema';
+import { UUID } from '@lumino/coreutils';
+import { startOfYesterday } from 'date-fns';
+import { useEffect, useState } from 'react';
+
+import useIsFirstRender from '@/src/shared/hooks/useIsFirstRender';
+import { products } from '@/src/stacBrowser/constants';
+import {
+  IStacItem,
+  IStacQueryBody,
+  IStacSearchResult,
+  StacFilterState,
+  StacFilterSetters,
+  StacFilterStateStateDb,
+} from '@/src/stacBrowser/types/types';
+import { GlobalStateDbManager } from '@/src/store';
+import { fetchWithProxies } from '@/src/tools';
+
+interface IUseStacSearchProps {
+  model: IJupyterGISModel | undefined;
+}
+
+interface IUseStacSearchReturn {
+  filterState: StacFilterState;
+  filterSetters: StacFilterSetters;
+  results: IStacItem[];
+  startTime: Date | undefined;
+  setStartTime: (date: Date | undefined) => void;
+  endTime: Date | undefined;
+  setEndTime: (date: Date | undefined) => void;
+  totalPages: number;
+  currentPage: number;
+  totalResults: number;
+  handlePaginationClick: (page: number) => Promise<void>;
+  handleResultClick: (id: string) => Promise<void>;
+  formatResult: (item: IStacItem) => string;
+  isLoading: boolean;
+}
+
+const API_URL = 'https://geodes-portal.cnes.fr/api/stac/search';
```

Comment:
Maybe for a future PR, we should think of making this user-provided. Maybe an `input` element in the UI to provide your endpoint for the STAC API. Or a setting.

We should also provide a way to add some API token if needed.